Applications/Tizen_CAPI/capi_func.c

```diff
@@ -299,75 +299,118 @@ int main(int argc, char *argv[]) {
 
   /* model create */
   status = ml_train_model_construct(&model);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    return 1;
+  }
+
   /* input layer create */
   status = ml_train_layer_create(&layers[0], ML_TRAIN_LAYER_TYPE_INPUT);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    NN_RETURN_STATUS();
+  }
 
   /* set property for input layer */
   status = ml_train_layer_set_property(layers[0], "input_shape=1:1:62720",
                                        "normalization=true",
                                        "bias_initializer=zeros", NULL);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    ml_train_layer_destroy(layers[0]);
+    NN_RETURN_STATUS();
+  }
 
   /* add input layer into model */
   status = ml_train_model_add_layer(model, layers[0]);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    ml_train_layer_destroy(layers[0]);
+    NN_RETURN_STATUS();
+  }
 
   /* create fully connected layer */
   status = ml_train_layer_create(&layers[1], ML_TRAIN_LAYER_TYPE_FC);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    ml_train_layer_destroy(layers[1]);
+    NN_RETURN_STATUS();
+  }
 
   /* set property for fc layer */
   status = ml_train_layer_set_property(
     layers[1], "unit= 10", "activation=softmax", "bias_initializer=zeros",
     "weight_regularizer=l2norm", "weight_regularizer_constant=0.005",
     "weight_initializer=xavier_uniform", NULL);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    ml_train_layer_destroy(layers[1]);
+    NN_RETURN_STATUS();
+  }
 
   /* add fc layer into model */
   status = ml_train_model_add_layer(model, layers[1]);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    ml_train_layer_destroy(layers[1]);
+    NN_RETURN_STATUS();
+  }
 
   /* create optimizer */
   status = ml_train_optimizer_create(&optimizer, ML_TRAIN_OPTIMIZER_TYPE_ADAM);
-  NN_RETURN_STATUS();
+  if (status != ML_ERROR_NONE) {
+    ml_train_optimizer_destroy(optimizer);
```

Comment:
destroy layers

Reply:
The same applies to all the error based exits below

Reply:
Done